src/libraries/Microsoft.PowerFx.Connectors/Public/CdpRecordType.cs

```diff
@@ -26,19 +26,39 @@ internal CdpRecordType(ConnectorType connectorType, DType recordType, ICdpTableR
             ReferencedEntities = referencedEntities;
         }
 
-        public override bool TryGetFieldType(string name, out FormulaType type)
+        public bool TryGetFieldExternalTableName(string fieldName, out string tableName)
         {
-            if (!base.TryGetBackingDType(name, out _))
+            if (!base.TryGetBackingDType(fieldName, out _))
+            {
+                tableName = null;
+                return false;
+            }
+
+            ConnectorType connectorType = ConnectorType.Fields.First(ct => ct.Name == fieldName);
```

Comment:
can fields be null? should we use FirstOrDefault?

Reply:
can we add test case where this fields is null as well?

Reply:
I understsand your concern but we test the field existence at line 31 so we're sure this line will work fine

Reply:
i see